### PR TITLE
W7 - anhquach - UI test

### PR DESCRIPTION
**Pull request by at-anhquach**
#### Related issues (Optional)
Task #84 

#### Description (Required)
- [x] Create Dialog
- [x] Create UI test

### CODE REVIEW

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import android.app.Instrumentation;
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+/**
+ * Created by anh.quach on 1/5/18.
+ * Test UI sign up fragment
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(AndroidJUnit4.class)
+public class SignUpUITest {
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [NoPackage] All classes and interfaces must belong to a named package
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#NoPackage

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        onView(withText(startsWith("Not enough data")))
+                .inRoot(withDecorView(not(is(mActivityRule.getActivity().getWindow().getDecorView()))))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkToastWhenNotDifferentUserName() {
+        setupFragment();
+        onView(withId(R.id.edtUsername))
+                .perform(typeText("hanangocanH"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "hanangocanH" appears 6 times in this file; the first occurrence is on line 55
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkToastWhenUsernameLessThan5OrMoreThan24() {
+        setupFragment();
+        onView(withId(R.id.edtUsername))
+                .perform(typeText("hana"), closeSoftKeyboard());
+        onView(withId(R.id.edtPassword))
+                .perform(typeText("ngocanh@TTJ"), closeSoftKeyboard());
```
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "ngocanh@TTJ" appears 5 times in this file; the first occurrence is on line 122
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/build.gradle [hunk 5]
```diff
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 26
     defaultConfig {
         applicationId "vn.asiantech.internship"
-        minSdkVersion 15
+        minSdkVersion 16
```
Comment (at-trungnguyen) [inline app/build.gradle]: why do you change this?

File: app/build.gradle [hunk 6]
```diff
@@ -94,18 +93,14 @@ dependencies {
...
     testImplementation 'org.mockito:mockito-core:2.13.0'
     // Android runner and rules support
     androidTestImplementation 'com.android.support.test:runner:1.0.1'
     androidTestImplementation 'com.android.support.test:rules:1.0.1'
-    //UI Espresso framework
-    androidTestImplementation'com.android.support.test.espresso:espresso-core:3.0.1'
+    // UI Espresso framework
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
     implementation 'junit:junit:4.12'
     androidTestCompile 'com.android.support.test.espresso:espresso-core:3.0.1'
```
Comment (at-trungnguyen) [inline app/build.gradle]: duplicate

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        onView(withId(R.id.edtUsername))
+                .perform(typeText("Hanangocanh"), closeSoftKeyboard());
+        onView(withId(R.id.edtPassword))
+                .perform(typeText("ngocanh@TTJ"), closeSoftKeyboard());
+        onView(withId(R.id.imgNext)).perform(click());
+        onView(withText("Done")).check(matches(isDisplayed()));
+    }
+
+    private void setupFragment() {
+        android.support.v4.app.FragmentManager fragmentManager = mActivityRule.getActivity().getSupportFragmentManager();
```
Comment (at-trungnguyen) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: `FragmentManager fragmentManager = mActivityRule.getActivity().getSupportFragmentManager();`

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java (hunk 1 above)
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [NoPackage] All classes and interfaces must belong to a named package
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#NoPackage

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java (hunk 3 above)
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "hanangocanH" appears 6 times in this file; the first occurrence is on line 55
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java (hunk 4 above)
Comment (at-dev-support-01) [inline app/src/androidTestAt-anhquachDebug/java/SignUpUITest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "ngocanh@TTJ" appears 5 times in this file; the first occurrence is on line 122
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/at-anhquach/java/vn/asiantech/internship/ui/login/SignUpFragment.java [hunk 8]
```diff
@@ -141,12 +146,55 @@ public void onResume() {
...
         } else {
             mImgNext.setSelected(false);
         }
     }
 
+    private void signUpValidation() {
+        mImgNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String username = mEdtUsername.getText().toString();
```
Comment (at-trungnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/ui/login/SignUpFragment.java]: `mEdtUsername.getText().toString().trim();`

File: app/src/at-anhquach/java/vn/asiantech/internship/ui/login/SignUpFragment.java [hunk 9]
```diff
@@ -141,12 +146,55 @@ public void onResume() {
...
+                String password = mEdtPassword.getText().toString();
```
Comment (at-trungnguyen) [inline app/src/at-anhquach/java/vn/asiantech/internship/ui/login/SignUpFragment.java]: `mEdtPassword.getText().toString().trim();`

**Pull request merged**